internal/client/client.go

```diff
@@ -1,15 +1,15 @@
-package client
-
-type ExchangeRate interface {
-	GetCurrentExchangeRate() (ExchangeRateResponse, error)
-}
-
-type Client struct {
-	ExchangeRate
-}
-
-func NewClient() *Client {
-	return &Client{
-		ExchangeRate: NewExchangeRateClient(),
-	}
-}
+package client
+
+type ExchangeRate interface {
+	GetCurrentExchangeRate() (ExchangeRateResponse, error)
+}
+
+type Client struct {
+	ExchangeRate
+}
+
+func NewClient() *Client {
+	return &Client{
+		ExchangeRate: NewExchangeRateClient(),
+	}
+}
```

Comment:
Such constructors make code tightly coupled. Here ExchangeRate cannot be configured outside if needed. I'd recommend don't create such structs (god objects - in this case client that aggregates many concrete type clients)

Reply:
Totally agree